test/functional/test_training.py

```diff
@@ -0,0 +1,91 @@
+# Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You
+# may not use this file except in compliance with the License. A copy of
+# the License is located at
+#
+#     http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is
+# distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+import json
+
+import pytest
+import sagemaker_containers.environment as environment
+from sagemaker_containers import Training
+
+
+@pytest.fixture(name='model_file')
+def fixture_model_file(opt_ml):
+    return str(opt_ml.join('output/saved_model.json'))
+
+
+@pytest.fixture(name='opt_ml')
+def create_opt_ml(tmpdir):
+    environment.base_dir = str(tmpdir)
+    tmpdir.mkdir('output')
+    tmpdir.mkdir('code')
+    return tmpdir
+
+
+def test_training_from_train_fn(opt_ml, model_file):
```

Comment:
Add a test to ensure that your class handles both a train_fn and a train_module being passed at the same time.

Reply:
refactored the classes so we do not have the edge case anymore.